packages/nextjs/components/Header.tsx

```diff
@@ -77,6 +78,25 @@ export const Header = () => {
   const { targetNetwork } = useTargetNetwork();
   const isLocalNetwork = targetNetwork.id === devnet.id;
 
+  const { provider } = useProvider();
+  const { address, status } = useAccount();
+  const [isDeployed, setIsDeployed] = useState(false);
+
+  useEffect(() => {
+    if (status === "connected" && address) {
+      provider
+        .getContractVersion(address)
+        .then((v) => {
+          if (v) setIsDeployed(true);
+        })
+        .catch((e) => {
+          console.log(e);
+          setIsDeployed(false);
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@metalboyrick

Reply:
in this case i think it's an ok exception, because you only want to watch `status` and `address` i presume. but if provider is not something that will be immediately available, then I would suggest adding a null check for that and add a watch on the dependency array, refer to my other review. (this way no eslint complaints either)